lib/rules/disallow-accessing-ember-properties.js

```diff
@@ -0,0 +1,33 @@
+var assert = require('assert');
+
+function DisallowAccessingEmberProperties() { }
+
+DisallowAccessingEmberProperties.prototype = {
+  configure: function(value) {
+    assert(
+      value === true,
+      this.getOptionName() + ' option requires a true value or should be removed'
+    );
+  },
+
+  getOptionName: function() {
+    return 'disallowAccessingEmberProperties';
+  },
+
+  check: function(file, errors) {
+    file.iterateNodesByType('MemberExpression', function(node) {
+      if (node.parentNode.type === 'AssignmentExpression') {
+        // Avoid throwing the error if we're assigning to an Ember property
+        return;
+      }
+      if (node.object.name === 'Ember') {
+        var propertyName = node.property.name;
+        errors.add(
```

Comment:
I think you should check for the assignment. Essentially, setting something on Ember is okay, such as the flag. Accessing properties or "constants" from Ember could still be done via destructuring first. We should probably think of various use cases though, and make sure there's nothing we're missing, but that's my immediate thought.